kernelci/config/rootfs.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2019 Collabora Limited
+# Author: Michal Galka <michal.galka@collabora.com>
+#
+# This module is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 2.1 of the License, or (at your option)
+# any later version.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+import yaml
+
+from kernelci.config import YAMLObject
+
+
+class RootFS(YAMLObject):
+    def __init__(self, name, rootfs_type):
+        self._name = name
+        self._rootfs_type = rootfs_type
+
+    @classmethod
+    def from_yaml(cls, rootfs, kw):
+        return cls(**kw)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def rootfs_type(self):
+        return self._rootfs_type
+
+
+class RootFS_Debos(RootFS):
+    def __init__(self, name, rootfs_type, debian_release=None,
+                 arch_list=None, extra_packages=None,
+                 extra_packages_remove=None,
+                 extra_files_remove=None, script=""):
+
+        super(RootFS_Debos, self).__init__(name, rootfs_type)
+        self._debian_release = debian_release
+        self._arch_list = arch_list or list()
+        self._extra_packages = extra_packages or list()
+        self._extra_packages_remove = extra_packages_remove or list()
+        self._extra_files_remove = extra_files_remove or list()
+        self._script = script
+
+    @classmethod
+    def from_yaml(cls, config, name):
+        kw = name
+        kw.update(cls._kw_from_yaml(
+            config, ['name', 'debian_release', 'arch_list',
+                     'extra_packages', 'extra_packages_remove',
+                     'extra_files_remove', 'script']))
+        return cls(**kw)
+
+    @property
+    def debian_release(self):
+        return self._debian_release
+
+    @property
+    def arch_list(self):
+        return list(self._arch_list)
+
+    @property
+    def extra_packages(self):
+        return self._extra_packages
```

Comment:
OK - except that by returning this list, it makes it possible to change it outside this class.  So maybe do like for `arch_list`:
```python
return list(self._extra_packages)
```